rmw_fastrtps_cpp/src/gid__type_support.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstddef>
+
+#include "rosidl_generator_c/message_type_support_struct.h"
+// rosidl_typesupport_cpp/message_type_support.hpp is installed by rosidl_generator_cpp
+#include "rosidl_typesupport_cpp/message_type_support.hpp"
+#include "rmw_dds_common/msg/gid__struct.hpp"
+#include "rmw_dds_common/msg/gid__rosidl_typesupport_fastrtps_cpp.hpp"
+#include "rmw_fastrtps_cpp/visibility_control.h"
+
+namespace rosidl_typesupport_cpp
+{
+
+template<>
+RMW_FASTRTPS_CPP_PUBLIC
+const rosidl_message_type_support_t *
+get_message_type_support_handle<rmw_dds_common::msg::Gid>()
+{
+  return ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME(
+    rosidl_typesupport_fastrtps_cpp, rmw_dds_common, msg, Gid)();
+}
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+RMW_FASTRTPS_CPP_PUBLIC
+const rosidl_message_type_support_t *
+ROSIDL_TYPESUPPORT_INTERFACE__MESSAGE_SYMBOL_NAME(
+  rosidl_typesupport_cpp, rmw_dds_common, msg, Gid)()
```

Comment:
I'm not calling that generator, because it introduces a circular dependency.

To be more specific, I avoided depending on `rosidl_typesupport_cpp`.
It has a build export dependency on [rmw_implementation](https://github.com/ros2/rosidl_typesupport/blob/4847322f6a03e5a2ec5cf0498ec8d93d0be8f7b4/rosidl_typesupport_cpp/package.xml#L25), which makes imposible to depend on that in `rmw_fastrtps_*`.
That implementation is needed because the generated typesupport code is dependent of the available typesupports,  and actually, of the typesupports [registered by the rmw implementations](https://github.com/ros2/rmw_fastrtps/blob/bd5507b61c7285d6083e514ccd49f00b2945baeb/rmw_fastrtps_cpp/CMakeLists.txt#L117).

----

If you have any suggestion to avoid duplication, it's welcomed.
But I didn't see a way of reusing that generator without generating a circular dependency, and I didn't want to also refactor how typesupports are generated to avoid the circular dependency (it would make this impossible to review).

Reply:
I see, I think that duplicating it in that case is ok, but please put a big notice about why that is there.

Reply:
You added a notice I presume?

Reply:
No, I didn't see your reply. Updating now.

Reply:
Updated in https://github.com/ros2/rmw_fastrtps/pull/312/commits/e6c618a6a40543800123c291fd2fd2ccb4ffd600